File: suii_task_manager/cppcode/src/TaskWithAction.cpp
#include "TaskWithAction.h"

TaskWithAction::TaskWithAction(void) : Task()
{
    this->setAction(-1);
}

void TaskWithAction::copyFromTask(Task* t)
{
    this->setType(t->type);
    this->setSrc(t->src);
    this->setDest(t->dest);
    this->setObj(t->obj);
    this->setContainer(t->container);
}

void TaskWithAction::setAction(int id)
{
    this->action = id;
    this->action_str = protocol.lookUpTaskActionID(id);
}

void TaskWithAction::printTaskData(void)
{
    std::cout << "Action: " << this->action_str << "; "; Task::printTaskData();
}

File: suii_task_manager/src/suii_task_manager/TaskWithAction.py
from Task import Task
from TaskProtocol import TaskProtocol

class TaskWithAction(Task):
    def __init__(self):
        Task.__init__(self)
        self.action = -1
        self.action_str = TaskProtocol.lookUpKey(TaskProtocol.taskActionDict, self.action)

    def setAction(self, id):
        self.action = id
        self.action_str = TaskProtocol.lookUpKey(TaskProtocol.taskActionDict, id)
    
    def copyFromTask(self, task):
        self.setType(task.type)
        self.setSrc(task.src)
        self.setDest(task.dest)
        self.setObj(task.obj)
        self.setContainer(task.container)
    
    def __repr__(self):
        return "<<'%s'>> %s" % (self.action_str, Task.__repr__(self))
